test/extensions/filters/common/expr/evaluator_fuzz_test.cc

```diff
@@ -0,0 +1,74 @@
+#include "common/network/utility.h"
+
+#include "extensions/filters/common/expr/evaluator.h"
+
+#include "test/common/stream_info/test_util.h"
+#include "test/extensions/filters/common/expr/evaluator_fuzz.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+#include "test/test_common/network_utility.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Expr {
+namespace {
+
+// Set stream_info using information from input.
+// Duplicates most of fromStreamInfo, except is more lightweight without the use of mocks.
+TestStreamInfo makeStreamInfo(const test::fuzz::StreamInfo& info) {
+  TestStreamInfo stream_info;
+  // TODO(asraa): Set upstreamHost() and downstreamSslConnection().
+  stream_info.metadata_ = info.dynamic_metadata();
+  const auto start_time =
+      static_cast<uint64_t>(std::numeric_limits<std::chrono::nanoseconds::rep>::max()) <
+              info.start_time()
+          ? 0
+          : info.start_time() / 1000;
+  stream_info.start_time_ = SystemTime(std::chrono::microseconds(start_time));
+  stream_info.setRequestedServerName(info.requested_server_name());
+  stream_info.response_code_ = info.has_response_code() ? info.response_code().value() : 200;
+  auto address = Network::Utility::resolveUrl("tcp://10.0.0.1:443");
```

Comment:
Why not fuzz the address?

Reply:
Done! I decided it might be good to fuzz the way we consume the address anyway.